### DISCUSSION TITLE
Entity doesn't get rendered

### DISCUSSION THREAD

**Discussion by MikeBrixia** [gurkenlabs/litiengine#789] [Q&A]
Hi!

I've recently started using LITIEngine and i was wondering how can i render entities sprites. I followed this tutorial from the docs [https://litiengine.com/docs/tutorials/creating-a-platformer/](url) with no results.

I setted up a quick test in utiliti where i have a level with a Player spawn and a creature with a sprite assigned to it.
![Utiliti](https://user-images.githubusercontent.com/96057936/235931104-bce4a7eb-b483-4299-9886-14ce4e62ad25.JPG)

When the game starts i call this code inside the game level class, trying to force the entity to be rendered
`

    public void onBeginPlay() {

        // Spawn the player at PlayerSpawn point location.
        Spawnpoint playerSpawn = levelEnvironment.getSpawnpoint("PlayerSpawn");
        playerSpawn.spawn(GameManager.getInstance().player);
       
        // This is where i tried to force the entity to be rendered.
        IEntity test = levelEnvironment.get("AI_Test");
        Game.graphics().renderEntity(GameManager.getInstance().gameScreen.getGraphicComponent(), test);
    }`

But the result is always the same:

![Utiliti](https://user-images.githubusercontent.com/96057936/235932705-f891284b-7a5d-40f7-9732-06a7be8d6264.JPG)

I'm not sure how the rendering part of the game is supposed to work. Placing creatures inside the level using utiliti editor is not enough? if so how do i manually render each of my entity?

**Comment by gattolfo98** [comment]
Same problem i don't understand very well how does Litiengine load textures theoretically based on the name of the sprite (name-state-direction) mine is called king-idle-right but it is not called if i switch to super("king") in my class player extends Creature

**Reply by MikeBrixia** [reply]
We're on the same page, after 3 weeks it would be great to at least be acknowledge

**Reply by gattolfo98** [reply]
> We're on the same page, after 3 weeks it would be great to at least be acknowledge

My solution was to change engine, this seems almost abandoned

**Reply by nightm4re94** [reply]
I'm sorry you feel this way, LITIENGINE is far from abandoned :)
It is, however, a project run by only two people, including development, distribution, documentation, community management, and user support. In all the years of providing this product for free, only very few community members have ever considered helping out with donations to keep development going. We are not even close to earning a single buck from developing this project, so please excuse us if we treat it like the spare time activity it is rather than a consumer-grade project with real customers.

**Comment by Hades948** [comment]
I don't use utiliti for much. There isn't really anything you _have_ to do in it that you can't just do in the Java code. Since I don't use utiliti much, I don't know exactly what your issue might be. But make sure you've called Environment.add() and pass your player instance. The gurkenlabs' GitHub page has a few projects that might be helpful: [https://github.com/gurkenlabs](https://github.com/gurkenlabs). Specifically, the ldjam ones have been helpful to me.

Also, keep in mind that the creators of the engine are busy people with real lives and might not always be available to respond. They also created the engine as a side project and aren't charging any money for it; they are not obligated to help anyone troubleshoot. Though, I've found they are generally pretty helpful, as is the rest of the community, on Discord. You might have better luck there. If you do, make sure to read through the rules and check out [https://dontasktoask.com/](https://dontasktoask.com/) (and the pages it links to at the bottom!) before asking your questions :) That's probably the best way to get a response.

**Comment by nightm4re94** [comment]
Hi, I'm sorry you had to wait this long for a response. 
Thanks @Hades948 for stepping in :)
It is usually not necessary to render Entities manually. Once an entity has been properly added to the Environment (and if your screen actually renders the environment), you should be able to see it. 
Could you please check if the debug renderer actually shows an entity on the environment?
To enable the debug renderer, modify the config.properties file with the following values:

- dbg_debugEnabled=true
- dbg_renderBoundingBoxes=true
- dbg_renderCollisionBoxes=true
